src/cmdlib.sh

```diff
@@ -392,11 +384,16 @@ runcompose() {
         sudo -E "$@"
         sudo chown -R -h "${USER}":"${USER}" "${tmprepo}"
     else
-        compose_qemu_args=()
-        if [ -f "${workdir}/cache/cache2.qcow2" ]; then
-            compose_qemu_args+=("-drive" "if=none,id=cache,discard=unmap,file=${workdir}/cache/cache2.qcow2" \
-                                "-device" "virtio-blk,drive=cache")
+        # "cache2" has an explicit label so we can find it in qemu easily
+        if [ ! -f "${workdir}"/cache/cache2.qcow2 ]; then
+            flock "${workdir}"/cache/cache2.lock -c "qemu-img create -f qcow2 cache2.qcow2.tmp 10G &&
```

Comment:
If we're going to do locking, shouldn't the check for existence also be under lock? Alternatively, we can just drop locking here since it's just used by `fetch` and `build` and those have to be in front of the pipeline and serialized.

Reply:
> If we're going to do locking, shouldn't the check for existence also be under lock? 

Yeah, fair.  As is this is like the "double checked locking" anti-pattern.

Reply:
I'd personally prefer to just drop locking entirely here. As is, the code is implying that we need locking because we do `runcompose` in parallel, which is not the case. This can be confusing to contributors.

If you want to keep it, maybe let's add a comment like "we don't technically need locking here but to be safe in case we do in the future".